### ISSUE TITLE
Calls made to random port (1128, 1129)

### ISSUE THREAD

**Issue by talibnawaz** [chequer-io/JDBC.NET#40]
Our Open Edge Progress database is behind a firewall and is accessible on port 10100. But regardless of giving the port number in the connection string, we have noticed that additional calls are being made to random ports, e.g. 1128 and 1129 in the screenshot below that are blocked by the firewall.

![image](https://github.com/chequer-io/JDBC.NET/assets/85462310/6fadceb1-ec25-41dd-8bd1-08d78db7727d)

Since the subsequent calls to random port numbers are blocked, we never get a response back.
Is there a way to control which port numbers will the library call?

**Comment by talibnawaz** [comment]
@tcables please help

**Comment by tcables** [comment]
@talibnawaz I'm sorry I'm not a primary developer, I just contributed once and answered a couple questions. I would recommend downloading the codebase, adding it to your solution, referencing it directly instead of the nuget package and stepthrough the code and see maybe where it could be coming from. otherwise if you do not find it, it could be the jar file. **BUT** I am not very familiar with this codebase. maybe @iodes or @evan-choi could shed some light if they have a moment

**Comment by chrisandchris** [comment]
@talibnawaz Did you ever got it working?

I'm struggling with properly configuring the Progress OpenEdge driver:

```
var builder = new JdbcConnectionStringBuilder
{
    DriverPath = "/Users/someuser/Downloads/openedge_jdbc_12.3/openedge.jar",
    DriverClass = "com.ddtek.jdbc.openedge.OpenEdgeDriver",
    JdbcUrl = "jdbc:datadirect:openedge://someip:someport;databaseName=somedb;user=someuser;password=somepassword",
}

using var connection = new JdbcConnection(builder);
connection.Open();
```

Just hangs on .Open() forever, no error appears. Do you have any experience with that one?

**Comment by talibnawaz** [comment]
@chrisandchris no, we had no luck with this one. We ended up going the ODBC route and using System.Data.Odbc was a huge relief over this one.

We looked up the issue I have mentioned in the original post, and it turned out to be an expected behaviour for it to make multiple connections to a list of ports

[Ports needed to be open - Progress Community](https://community.progress.com/s/article/P98541)

**Comment by talibnawaz** [comment]
> [@talibnawaz](https://github.com/talibnawaz) Did you ever got it working?
> 
> I'm struggling with properly configuring the Progress OpenEdge driver:
> 
> ```
> var builder = new JdbcConnectionStringBuilder
> {
>     DriverPath = "/Users/someuser/Downloads/openedge_jdbc_12.3/openedge.jar",
>     DriverClass = "com.ddtek.jdbc.openedge.OpenEdgeDriver",
>     JdbcUrl = "jdbc:datadirect:openedge://someip:someport;databaseName=somedb;user=someuser;password=somepassword",
> }
> 
> using var connection = new JdbcConnection(builder);
> connection.Open();
> ```
> 
> Just hangs on .Open() forever, no error appears. Do you have any experience with that one?

The only difference I see between my code and yours is a semicolon missing in your JDBC URL after the password. Not sure if it makes any difference but might be worth a try
